### PR TITLE
Fix role granting to not loose existing privileges.

### LINKED ISSUE #16710: Granting a role to a user wipes all direct privileges
### CrateDB version

5.8.3

### CrateDB setup information

Single node local

### Problem description

The privileges granted directly to a user should take precedence over those inherited from a role.
After granting a role to a user who already has direct ALL privileges, querying the sys.users table results in a `SchemaUnknownException`. This behavior is unexpected because direct privileges should take precedence over role-based privileges.

### Steps to Reproduce

**crate**
```
~/ crash
cr> CREATE USER my_user WITH (password='my_password');
cr> GRANT ALL TO my_user;

```

**my_user**
```
~/ crash -U my_user -W
cr> SELECT current_user;
+--------------+
| CURRENT_USER |
+--------------+
| my_user      |
+--------------+
cr> SELECT * FROM sys.users;
+---------------+------+---------+----------+-----------+
| granted_roles | jwt  | name    | password | superuser |
+---------------+------+---------+----------+-----------+
| []            | NULL | crate   | NULL     | TRUE      |
| []            | NULL | my_user | ******** | FALSE     |
+---------------+------+---------+----------+-----------+
```

**crate**
```
cr> CREATE ROLE my_role2;
cr> GRANT my_role2 TO my_user;
```

**my_user**
```
cr> SELECT * FROM sys.users;
SchemaUnknownException[Schema 'sys' unknown]
```

### Actual Result

After assigning the role my_role2 to my_user, attempting to query sys.users results in a SchemaUnknownException.

### Expected Result

The user my_user should continue to have access to the sys.users table since they have direct ALL privileges.

### PR DESCRIPTION
**Pull request by seut**
If a role is granted of another role, it's privileges must remain.

Fixes #16710.

### CODE REVIEW

File: docs/appendices/release-notes/5.8.4.rst [hunk 1]
```diff
@@ -108,3 +108,6 @@ Fixes
 
 - Fixed an issue which may cause a ``EXPLAIN ANALYZE`` to throw exception due
   to internal concurrent unsafe access.
+
+- Fixed an issue causing a role to loose it's concrete privileges once it gets
+  granted of another role.
```
Comment (matriv) [inline docs/appendices/release-notes/5.8.4.rst:113]: I think it reads better just like:
```suggestion
  granted another role.
```

**Review by matriv — APPROVED** [review]
LGTM. thank you!
(left minor wording suggestion)

File: docs/appendices/release-notes/5.8.4.rst (hunk 1 above)
Comment (seut) [reply docs/appendices/release-notes/5.8.4.rst]: Thanks applied, also added a link to the `GRANT` statement.

Comment (matriv) [reply docs/appendices/release-notes/5.8.4.rst]: nice, thx!

**Pull request merged**